### PR TITLE
DD-463

### PR DESCRIPTION
**Pull request by dldevbot**
Re-enabling graphql integration test.

### CODE REVIEW

File: opencti-platform/opencti-graphql/Dockerfile [hunk 1]
```diff
@@ -1,50 +1,32 @@
...
 FROM base AS graphql-builder
 WORKDIR /opt/opencti-build/opencti-graphql
-COPY ./package.json ./yarn.lock ./.yarnrc.yml ./
-COPY ./.yarn ./.yarn
-RUN yarn install --network-timeout 300000
-COPY . /opt/opencti-build/opencti-graphql
+COPY . .
+#Missing node-gyp as a dep?  Usage Error: Couldn't find a script name "node-gyp" in the top-level (used by deasync@npm:0.1.28)
+#I've seen it build before.
+RUN yarn add node-gyp
```
Comment (r-joyce) [inline opencti-platform/opencti-graphql/Dockerfile]: This is a no no, this is adding a dependency to our package.json file at build time

File: opencti-platform/opencti-graphql/Dockerfile-test [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    rm -rv /usr/lib/python*/ensurepip; \
+    pip3 install --no-cache-dir --upgrade pip setuptools wheel; \
+    ln -sf python3 /usr/bin/python;
+
+WORKDIR /opt/test
+COPY . .
+RUN yarn add node-gyp
+RUN yarn install --check-cache --inline-builds --network-timeout 300000 --immutable
+
+CMD ["yarn", "test"]
```
Comment (r-joyce) [inline opencti-platform/opencti-graphql/Dockerfile-test]: I don't think we should commit this file. It's fine if you wanna keep a local copy for your own testing, but developers working on this should be able to run tests locally without the need of Docker and the build servers should already be wrapping in a container during builds/tests

File: opencti-platform/opencti-graphql/package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "lint": "eslint --max-warnings 0 .",
     "schema-compile": "node schema_merge.js",
     "migrate:add": "migrate create --template-file src/utils/migration-template.js --migrations-dir=./src/migrations",
-    "test": "jest --bail --testSequencer ./tests/utils/testSequencer.js --verbose --runInBand --forceExit --coverage --e test",
+    "test": "jest --bail --testSequencer ./tests/utils/testSequencer.js --verbose --runInBand --forceExit --coverage --e test --testPathIgnorePatterns=\"tests/cyio/integrity\" --testPathIgnorePatterns=\"tests/0.*\"",
```
Comment (r-joyce) [inline opencti-platform/opencti-graphql/package.json]: Would you mind instead moving these changes into `opencti-platform/opencti-graphql/jest.config.js`? There should already be a field in there for `testPathIgnorePatterns`

File: opencti-platform/opencti-graphql/tests/cyio/config.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ const buildHeaders = () => {
 };
 
 export default async function submitOperation(operation, vars, res) {
-  return new Promise((resolve) => {
+  return new Promise((resolve,reject) => {
```
Comment (r-joyce) [inline opencti-platform/opencti-graphql/tests/cyio/config.js:16]: Nice catch, yeah there's def an edge case in logic here that won't lead to a new Promise being created. Not sure if this is the correct way to handle it, but I'll dig into this further.

File: opencti-platform/opencti-graphql/Dockerfile [hunk 5]
```diff
@@ -1,50 +1,32 @@
...
-    rm -rv /usr/lib/python*/ensurepip; \
-    pip3 install --no-cache-dir --upgrade pip setuptools wheel; \
-    ln -sf python3 /usr/bin/python;
 WORKDIR /opt/opencti
 COPY ./src/python/requirements.txt ./src/python/requirements.txt
 RUN pip3 install --no-cache-dir --requirement ./src/python/requirements.txt
 RUN pip3 install --upgrade --force --no-cache-dir git+https://github.com/OpenCTI-Platform/client-python@master
-RUN apk del git python3-dev gcc musl-dev
-COPY --from=graphql-deps-builder /opt/opencti-build/opencti-graphql/node_modules ./node_modules
+RUN apk del git python3-dev gcc musl-dev make g++
```
Comment (r-joyce) [inline opencti-platform/opencti-graphql/Dockerfile]: These shouldn't be needed...

**Review by r-joyce — CHANGES_REQUESTED** [review]
We should be able to get tests running without modifying our current Dockerfile, I think by making those changes we might inadvertently add more complexity at this moment.

File: opencti-platform/opencti-graphql/Dockerfile-test (hunk 2 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile-test]: Similar to encapsulating the build entirely within a container, I would like to do similar for testing to reduce the likelihood that environmental differences cause difference in behavior.

File: opencti-platform/opencti-graphql/package.json (hunk 3 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/package.json]: Roger.

File: opencti-platform/opencti-graphql/tests/cyio/config.js (hunk 4 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/tests/cyio/config.js:16]: I tried to keep the code as is as much as possible. I didn't find exact root cause as to why the tests would hang if either of the TESTING_ env vars weren't provided.  This fix seemed reasonable and appears to work.

Do you want to keep this PR open until we find a root cause or just create a separate bug ticket in Jira?

File: opencti-platform/opencti-graphql/Dockerfile (hunk 5 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile]: I kept the code as similar to the original as possible.  By not needed do we mean the packages referenced in apk del -- as in they shouldn't be installed in the 1st place?  Also node_modules shouldn't be needed at all in the final image?

Comment (r-joyce) [reply opencti-platform/opencti-graphql/Dockerfile]: Oh weird I had specific lines highlighted when I did the review. For this one, I'm specifically referring to the adding of `make g++`, we can build a production ready image now without those.

File: opencti-platform/opencti-graphql/Dockerfile-test (hunk 2 above)
Comment (r-joyce) [reply opencti-platform/opencti-graphql/Dockerfile-test]: Yeah I agree, but it should be happening within the Jenkinsfile. We shouldn't need another Dockerfile, and the local dependencies should be resolved when we are already doing a build, so we really only need the `yarn test` line out of all of this. We can hop on a call and I can go over more about how this is done in a Jenkinsfile.

File: opencti-platform/opencti-graphql/tests/cyio/config.js (hunk 4 above)
Comment (r-joyce) [reply opencti-platform/opencti-graphql/tests/cyio/config.js:16]: We can continue committing changes to this branch
And I guess I should have explained that both of those env variables are required to run tests, and are expected to be there when we run them

File: opencti-platform/opencti-graphql/Dockerfile (hunk 1 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile]: [UPDATE: my comment didn't show up for some reason] 

Should I add this to the package.json then?  

For context, my Desktop wasn't able to find/build node-gyp.  In the past I was able to get node-gyp to build during yarn install, but not sure why this is no longer the case.

File: opencti-platform/opencti-graphql/Dockerfile-test (hunk 2 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile-test]: Jenkinsfile still leaves it up to outside container aka inconsistent environment -- eg one dev will say "but it works for me" but then Jenkins can't run it.
Ya, it's redundant wrt the build.  We can run it from the build container?

File: opencti-platform/opencti-graphql/tests/cyio/config.js (hunk 4 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/tests/cyio/config.js:16]: Ya, I wanted to prevent Jenkins from hanging and also anyone new coming onboard from not figuring out what the issue was right away -- this is at least fail fast.

File: opencti-platform/opencti-graphql/Dockerfile (hunk 5 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile]: I wasn't able to - my dev env doesn't work so I put everything in the container.  To build dependencies I needed to add g++ and make... that being said, it's still an inconsistent build -- I had to re-add node-gyp.

File: opencti-platform/opencti-graphql/Dockerfile (hunk 1 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile]: working build, within container: added node-gyp 9.3.1 as dev dep in package.json, installed build tools make & g++, yarn install without --immutable
`corepack enable`
`apk add --no-cache git tini gcc musl-dev python3 python3-dev postfix postfix-pcre make g++`
`python3 -m ensurepip`
`pip3 install --no-cache-dir --upgrade pip setuptools wheel`
`ln -sf python3 /usr/bin/python`
`yarn install --check-cache --inline-builds --network-timeout 300000`

Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile]: unfortunately the deasync that gets created appears to be arch specific

➤ YN0000: │ deasync@npm:0.1.28 STDOUT Installed in `/test/node_modules/deasync/bin/linux-arm64-node-16/deasync.node`

File: opencti-platform/opencti-graphql/Dockerfile (hunk 5 above)
Comment (dldevbot) [reply opencti-platform/opencti-graphql/Dockerfile]: update: agreed to revert Dockerfile so it matches opencti official.  will wait for updates upstream that support arm64 arch.

**Review by r-joyce — APPROVED** [review]
